subsys/bluetooth/host/keys_br.c

```diff
@@ -85,3 +127,112 @@ void bt_keys_link_key_clear_addr(const bt_addr_t *addr)
 		bt_keys_link_key_clear(key);
 	}
 }
+
+void bt_keys_link_key_store(struct bt_keys_link_key *link_key)
+{
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		int err;
+		char key[BT_SETTINGS_KEY_MAX];
+		bt_addr_le_t le_addr;
+
+		le_addr.type = BT_ADDR_LE_PUBLIC;
+		bt_addr_copy(&le_addr.a, &link_key->addr);
+		bt_settings_encode_key(key, sizeof(key), "link_key",
+			&le_addr, NULL);
+
+		err = settings_save_one(key, link_key->storage_start,
+			BT_KEYS_LINK_KEY_STORAGE_LEN);
+		if (err) {
+			BT_ERR("Failed to svae link key (err %d)", err);
+		}
+	}
+}
+
+#if defined(CONFIG_BT_SETTINGS)
+
+static int link_key_set(const char *name, size_t len_rd,
+	settings_read_cb read_cb, void *cb_arg)
+{
+	int err;
+	ssize_t len;
+	bt_addr_le_t le_addr;
+	struct bt_keys_link_key *link_key;
+	char val[BT_KEYS_LINK_KEY_STORAGE_LEN];
+
+	if (!name) {
+		BT_ERR("Insufficient number of arguments");
+		return -EINVAL;
+	}
+
+	len = read_cb(cb_arg, val, sizeof(val));
+	if (len < 0) {
+		BT_ERR("Failed to read value (err %zu)", len);
+		return -EINVAL;
+	}
+
+	BT_DBG("name %s val %s", log_strdup(name),
+		(len) ? bt_hex(val, sizeof(val)) : "(null)");
+
+	err = bt_settings_decode_key(name, &le_addr);
+	if (err) {
+		BT_ERR("Unable to decode address %s", name);
+		return -EINVAL;
+	}
+
+	link_key = bt_keys_get_link_key(&le_addr.a);
+	if (len != BT_KEYS_LINK_KEY_STORAGE_LEN) {
+		if (link_key) {
+			bt_keys_link_key_clear(link_key);
+			BT_DBG("Clear keys for %s", bt_addr_le_str(&le_addr));
+		} else {
+			BT_WARN("Unable to find deleted keys for %s",
+				bt_addr_le_str(&le_addr));
+		}
+
+		return 0;
+	}
+
+	memcpy(link_key->storage_start, val, len);
+	BT_DBG("Successfully restored link key for %s",
+		bt_addr_le_str(&le_addr));
+#if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (aging_counter_val < link_key->aging_counter) {
+		aging_counter_val = link_key->aging_counter;
+	}
+#endif  /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
+
+	return 0;
+}
+
+static int link_key_commit(void)
+{
+	return 0;
+}
+
+SETTINGS_STATIC_HANDLER_DEFINE(bt_link_key, "bt/link_key", NULL, link_key_set,
+	link_key_commit, NULL);
+
+void bt_keys_link_key_update_usage(const bt_addr_t *addr)
+{
+	struct bt_keys_link_key *link_key = bt_keys_find_link_key(addr);
+
+	if (!link_key) {
+		return;
+	}
+
+	if (last_keys_updated == link_key) {
+		return;
+	}
+
+	link_key->aging_counter = ++aging_counter_val;
+	last_keys_updated = link_key;
+
+	BT_DBG("Aging counter for %s is set to %u", bt_addr_str(addr),
+	       link_key->aging_counter);
```

Comment:
Here you got it right - are you doing this manually? Most decent editors should be possible to configure to do it automatically for you so you don't have to think about it.